projects/reth-pureth-meta.md

```diff
@@ -0,0 +1,152 @@
+    gas_used,
+    timestamp,
+    extradata,
+    prev_randao,
+    0x0000000000000000, # nonce
+    base_fee_per_gas,
+    withdrawals_root,
+    blob_gas_used,
+    excess_blob_gas,
+    parent_beacon_block_root,
+])
+```
+
+For now, this can be done by making the logsBloom field's first 256 bits the log index root hash and then appending it with zeroes. 
+
+#### Execution Layer Changes
+Some new data structures have to be defined for the Execution layer:
+```rust
+#[derive(Encode, Decode)]
+struct LogIndex {
+    epochs: Vec<LogIndexEpoch>,
+    next_index: u64,
+}
+
+#[derive(Encode, Decode)]
+struct LogIndexEpoch {
+    filter_maps: Vec<Vec<FilterRow>>,
+    log_entries: Vec<LogEntry>,
+}
+
+type FilterRow = ProgressiveByteList[(MAX_BASE_ROW_LENGTH * log2(MAP_WIDTH)).div_floor(8 * MAPS_PER_EPOCH), LAYER_COMMON_RATIO]
+
+
+#[derive(Encode, Decode)]
+struct LogEntry {
+    log: Log,
+    meta: LogMeta,
+}
+
+#[derive(Encode, Decode)]
+struct LogMeta {
+    block_number: u64,
+    transaction_hash: B256,
+    transaction_index: u64,
+    log_in_tx_index: u64,
+}
+
+
+#[derive(Encode, Decode)]
+struct Log {
+    address: Address,
+    topics: Vec<Bytes>,
+    data: Vec<Bytes>,
+}
+
+#[derive(Encode, Decode)]
+struct BlockDelimiterEntry{
+    dummy_log: Log,        // zero address and empty lists
+    meta: BlockDelimiterMeta,
+}
+
+#[derive(Encode, Decode)]
+struct BlockDelimiterMeta{
+    block_number: u64,
+    block_hash: B256,
+    timestamp: u64,
+    dummy_value: u64,  // 2**64-1
+}
+
+```
+
+
+These structures also require implementing the [Progressive List](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7916.md) SSZ structure, which for now can replaced by `List[type, 9999999]`, which would be trivial to change afterwards.
+
+The EIP also requires to implement proofs for the log values and their block ranges.
+
+Also the RPC endpoints `eth_getLogs`, `eth_newFilter`, `eth_getFilterLogs`, `eth_getFilterChanges` have to be implemented for the new data structure.
+
+### Roadmap
+#### Milestone 0: Week 6
+- Get familiar with Reth codebase and figure out the parts of the codebase to change
+
+#### Milestone 1: Week 7-10
+- Implement basic containers and structures in Reth
+- Lay down a basic structure/framework to work with
+
+#### Milestone 2: Week 10-15
+- Implement function for inserting logs, retrieving logs and understand how the logs proof works in detail
+- Test basic insertions and retrievals of logs
+
+#### Milestone 3: Week 16-18
+- Implement proof of inclusions, completeness proofs and block range proofs
+
+#### Milestone 4: Week 19-22
+- Test whether proofs work correctly and implement RPC endpoints to fetch logs and proofs
+
+#### Milestone 5: Week 23 onwards
+- Complete spec tests, local interop testing
+- Test with kurtosis,  hive and other tooling
+- Test RPC endpoints and valid and invalid proofs
+
+## Possible Challenges
+- Complexity of log filter structure
+- Complexity of proof generation and verification
```

Comment:
you could elaborate on this, what does this mean - problems with debugging, long implementation time..? Any mitigations?